packages/auth/src/authClients/clerk.ts

```diff
@@ -0,0 +1,27 @@
+import { Clerk } from '@clerk/clerk-js'
+import { UserResource as ClerkUserResource } from '@clerk/types'
+
+import type { AuthClient } from '.'
+
+export type AuthClientClerk = AuthClient
+
+export type { Clerk }
+
+export type ClerkUser = ClerkUserResource & { roles: string[] | null }
+
+export const clerk = (client: Clerk): AuthClientClerk => {
+  return {
+    type: 'clerk',
+    client,
+    login: async (options?) => client.openSignIn(options || {}),
+    logout: async () => new Promise((res) => client.signOut(res)),
+    signup: async (options?) => client.openSignUp(options || {}),
+    // Clerk uses the session ID PLUS the __session cookie.
+    getToken: async () => client.session.id,
+    getUserMetadata: async () => {
+      return client.user
+        ? { ...client.user, roles: client.user.publicMetadata['roles'] ?? [] }
```

Comment:
```suggestion
        ? { ...client.user, roles: client.user?.publicMetadata?.['roles'] ?? [] }
```

Reply:
Fine adding the latter one - Clerk's typing claims it will always be defined but I should know better than to blindly trust typing. The first `?` is unnecessary, I think, since we're in a "that value is truthy" branch here already.